client/llb/source.go

```diff
@@ -373,6 +380,12 @@ func MountSSHSock(sshID string) GitOption {
 	})
 }
 
+func GitChecksum(v string) GitOption {
```

Comment:
Can we connect this with `ADD --checksum` already in this PR in case we take this separately from #5974 ?

Reply:
done